l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -145,10 +144,10 @@ contract ExecutorFacet is Base, IExecutor {
                 require(logSender == L2_TO_L1_MESSENGER_SYSTEM_CONTRACT_ADDR, "lm");
                 l2LogsTreeRoot = logValue;
             } else if (logKey == uint256(SystemLogKey.TOTAL_L2_TO_L1_PUBDATA_KEY)) {
```

Comment:
I'm not an expert in the protocol, but this `if` statement looks like it should be processed just once.
Is that the case? If yes, would it make sense to move the calculation of `providedL2ToL1PubdataHash ` inside this statement, so that we don't have an `if` on the mode twice?

Reply:
Actually, Vlad's suggestion is better.